test/unit/org/apache/cassandra/tools/ToolRunner.java

```diff
@@ -330,12 +330,21 @@ public ToolRunner waitAndAssertOnExitCode()
     
     public ToolRunner waitAndAssertOnCleanExit()
     {
-        return waitAndAssertOnExitCode().assertEmptyStdErr();
+        return waitAndAssertOnExitCode().assertCleanStdErr();
     }
     
-    public ToolRunner assertEmptyStdErr()
+    /**
+     * Checks if the stdErr is empty after removing any JVM env info output
+     * 
+     * Some JVM configs may output env info on stdErr. We need to remove those to see what was the tool's actual stdErr
+     * 
+     * @return
+     */
+    public ToolRunner assertCleanStdErr()
```

Comment:
Thanks for this!

To help make this more extendable, it would be good to use a exclude list and filter based off that; this is what happens in python dtests (example: https://github.com/riptano/ccm/blob/master/ccmlib/node.py#L395-L421) today.

With an exclude list, we can add more cases by default over time, but can also let tests control and search for the logs they care about.

Reply:
Aha! icywm. Iiuc my latest commit works off an exclude list. I haven't gone as far as implementing the `grepLog` functionality as it sounds more like sthg for another ticket + I want to merger this asap bc of 15583 (the main tooling ticket).

Still I think this addition is nice and a step in the direction you point. Wdyt?